backend/src/database/querys.rs

```diff
@@ -0,0 +1,173 @@
+use crate::{
+    database::dto::{AttestationResponse, Pagination},
+    error::AppError,
+};
+
+use sqlx::{postgres::PgQueryResult, Execute, FromRow, PgPool, Postgres, QueryBuilder};
+use uuid::Uuid;
+
+use super::dto::{AttestationRequest, Credential};
+
+pub async fn get_attestation_request_by_id(
+    attestation_request_id: &Uuid,
+    db_executor: &PgPool,
+) -> Result<AttestationResponse, AppError> {
+    sqlx::query_as!(
+        AttestationResponse,
+        "SELECT * FROM attestation_requests WHERE id = $1",
+        attestation_request_id
+    )
+    .fetch_one(db_executor)
+    .await
+    .map_err(AppError::from)
+}
+
+pub async fn get_attestations_count(db_executor: &PgPool) -> i64 {
+    sqlx::query_scalar!("SELECT COUNT (*) FROM attestation_requests")
+        .fetch_one(db_executor)
+        .await
+        .map_or(0, |count| count.unwrap())
+}
+
+pub async fn get_attestation_requests(
+    pagination: Pagination,
+    db_executor: &PgPool,
+) -> Result<Vec<AttestationResponse>, AppError> {
+    let query = build_pagination_query(pagination);
+    get_attestations(&query, db_executor)
+        .await
+        .map_err(AppError::from)
+}
+
+fn build_pagination_query(pagination: Pagination) -> String {
+    let mut query: QueryBuilder<Postgres> =
+        QueryBuilder::new("SELECT * FROM attestation_requests WHERE deleted_at IS NULL ");
+
+    if let Some(sorting) = pagination.sort {
+        query.push(format!("ORDER BY {} {} ", sorting[0], sorting[1]));
```

Comment:
classic SQL injection!!!! Just pass something like "; DROP TABLE attestations" as sort parameter and it will blow up.

Reply:
https://sqlx.dev/article/SQLX_Security_Protecting_Your_Data_from_SQL_Injection_Attacks.html

Reply:
... no my backdor :ghost:. removed in: [2f49949](https://github.com/KILTprotocol/attester/pull/1/commits/2f49949a05b2edff333825be7b098c6ceb6037a0)